### PR TITLE
 Modernize plugin

### PR DESCRIPTION
**Pull request by casz**
* Simplify Jelly
* 🙈 Simplify plugin by deprecating "other" fields as they were poorly implemented.
* Implement SimpleBuildStep for pipeline support
* Add Symbol for better pipeline support 😄

`vsTest()` 👍 

image

### CODE REVIEW

**Comment by casz** [comment]
@egoughnour defaultCharset does the detection and otherwise defaults to UTF-8

**Comment by egoughnour** [comment]
@casz that leaves just (1) and (2).  They are connected, in that if the user can configure the plugin for .NET Standard, Core, etc., they may not be running Windows.   I am not sure that is a legitimate use case though.

File: src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestInstallation.java [hunk 1]
```diff
@@ -61,8 +68,63 @@ protected Object readResolve() {
...
+        return installations;
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) Jenkins.getInstance().getDescriptorOrDie(getClass());
+    }
+
+    private static boolean isWindows() {
+        return File.pathSeparatorChar == ';';
```
Comment (casz) [inline src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestInstallation.java]: This is meant for Jenkins Master to be able to default whether ".exe" or without. So if the Jenkins master is running Linux

**Comment by casz** [comment]
1. Currently you can do both, the combobox allows free form as well as enumerations.
Not sure where 2. fits in. Could you provide more details?

**Comment by egoughnour** [comment]
I think I have already sorted that out. I wasn't sure if there was a case where the Jenkins _Agent_ might be on Linux.
If the user were testing cross-platform binaries on a more recent .NET Framework that could almost be feasible. However I think that would mean the user had already re-implemented, say, the VS 2017 Test agent, for instance.

**Comment by casz** [comment]
To release I suggest using the `mvn release:prepare release:perform` to release
Please read https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins#HostingPlugins-Releasingtojenkins-ci.org for how it works. This will remove the `-SNAPSHOT` make a commit than release and then bump version with `-SNAPSHOT` and commit that

**Pull request merged**